src/main/java/com/influxdb/v3/client/write/WriteOptions.java

```diff
@@ -96,6 +120,17 @@ public WritePrecision precisionSafe(@Nonnull final ClientConfig config) {
                 : (config.getWritePrecision() != null ? config.getWritePrecision() : DEFAULT_WRITE_PRECISION);
     }
 
+    /**
+     * @param config with/without defaultTags defined
+     * @return defaultTags - can be an empty map if none are defined.
+     */
+    @Nonnull
+    public Map<String, String> defaultTagsSafe(@Nonnull final ClientConfig config) {
+        Arguments.checkNotNull(config, "config");
+        return defaultTags.size() > 0 ? defaultTags
```

Comment:
Instead of the size we can use `isEmpty`:

```suggestion
        return !defaultTags.isEmpty() ? defaultTags
```

Reply:
Changed to 

```java
        return defaultTags.isEmpty()
          ? (config.getDefaultTags() != null
              ? config.getDefaultTags()
              : defaultTags
            )
          : defaultTags;
```